homeassistant/components/zwave/__init__.py

```diff
@@ -376,6 +376,24 @@ def _on_timeout(sec):
         hass.add_job(check_has_unique_id, entity, _on_ready, _on_timeout,
                      hass.loop)
 
+    def node_removed(node):
+        node_id = node.node_id
+        node_key = 'node-{}'.format(node_id)
```

Comment:
We should extract this into a Z-Wave helper to guarantee we always generate the same key.

Reply:
There's a couple existing functions to generate and return a unique_id, but there's a decent amount of refactoring that would be needed to make them usable from within node_removed since we don't have access to the same data they expect. Should I try to refactor that for this PR or save it for a future PR?

Reply:
future PR.

Reply:
Not sure what other data is expected? I really just want a function that generates the key off a node. 

```
def generate_key(node_id):
    return "node-{}".format(node_id)
```

Reply:
From what I can tell looking through the code, the existing function needs the entire ZWaveNode object from OZW and does some additional checks to see if it has been fully parsed/loaded by OZW before issuing a unique_id based on the node ID.

Reply:
But that logic is not needed here?